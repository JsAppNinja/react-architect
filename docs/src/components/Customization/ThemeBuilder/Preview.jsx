import React, { PureComponent } from 'react';
import { Button, Grid, Cell, Toolbar, Drawer, bem } from 'react-md';

const NAV_ITEMS = [
  { primaryText: 'Woop woop' },
  { primaryText: 'That\'s the sound' },
  { primaryText: 'Of the Police' },
];

const idPrefix = 'theme-builder-preview';

export default class Preview extends PureComponent {
  state = { visible: false };

  toggleDrawer = () => {
    this.setState({ visible: !this.state.visible });
  };

  handleVisibilityChange = (visible) => {
    this.setState({ visible });
  };

  render() {
    return (
      <section className="md-background--card theme-preview" ref={(container) => { this._container = container; }}>
        <Toolbar
          id={`${idPrefix}-toolbar`}
          nav={<Button id={`${idPrefix}-drawer-toggle`} icon onClick={this.toggleDrawer}>menu</Button>}
          title="Theme Preview"
          colored
          fixed
        />
        <Grid className="md-toolbar-relative">
          <Cell component="h2" size={12} className="md-display-1">Look at this</Cell>
          <Button id={`${idPrefix}-btn`} primary raised className={bem('theme-preview', 'btn')}>
            Button
          </Button>
        </Grid>
        <Drawer
          id={`${idPrefix}-drawer`}
          renderNode={this._container}
          header={<Toolbar title="Theme Preview" id={`${idPrefix}-drawer-toolbar`} />}
          navClassName="md-toolbar-relative"
          navItems={NAV_ITEMS}
          visible={this.state.visible}
          onVisibilityChange={this.handleVisibilityChange}
          overlay
          type={Drawer.DrawerTypes.TEMPORARY}
        />
        <Button id={`${idPrefix}-floating-btn`} floating secondary fixed>email</Button>
      </section>
    );
  }
}
