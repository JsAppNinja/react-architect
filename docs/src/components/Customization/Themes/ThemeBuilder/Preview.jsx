import React, { PureComponent } from 'react';
import Button from 'react-md/lib/Buttons/Button';
import Toolbar from 'react-md/lib/Toolbars';
import Drawer from 'react-md/lib/Drawers';

const NAV_ITEMS = [
  { primaryText: 'Woop woop' },
  { primaryText: 'That\'s the sound' },
  { primaryText: 'Of the Police' },
];

export default class Preview extends PureComponent {
  state = { visible: false };

  toggleDrawer = () => {
    this.setState({ visible: !this.state.visible });
  };

  handleVisibilityChange = (visible) => {
    this.setState({ visible });
  };

  render() {
    return (
      <section className="md-background--card theme-preview" ref={(container) => { this._container = container; }}>
        <Toolbar
          nav={<Button icon onClick={this.toggleDrawer}>menu</Button>}
          title="Theme Preview"
          colored
          fixed
        />
        <div className="md-grid md-toolbar-relative">
          <h2 className="md-display-1 md-cell md-cell--12">Look at this</h2>
          <Button primary raised className="theme-preview__btn">
            Button
          </Button>
        </div>
        <Drawer
          renderNode={this._container}
          header={<Toolbar title="Theme Preview" />}
          navClassName="md-toolbar-relative"
          navItems={NAV_ITEMS}
          visible={this.state.visible}
          onVisibilityChange={this.handleVisibilityChange}
          overlay
          type={Drawer.DrawerTypes.TEMPORARY}
        />
        <Button floating secondary fixed>email</Button>
      </section>
    );
  }
}
