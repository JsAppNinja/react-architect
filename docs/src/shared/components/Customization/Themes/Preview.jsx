import React, { PureComponent, PropTypes } from 'react';
import Button from 'react-md/lib/Buttons/Button';
import Toolbar from 'react-md/lib/Toolbars';
import Drawer from 'react-md/lib/Drawers';

import './_styles.scss';
import LoremIpsum from 'components/LoremIpsum';
const NAV_ITEMS = [{ primaryText: 'Woop Woop' }, { primaryText: 'That\'s the Sound' }, { primaryText: 'Of the Police' }];

export default class Preview extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
  };

  constructor(props) {
    super(props);

    this.state = { visible: false };
  }

  _toggleDrawer = () => {
    this.setState({ visible: !this.state.visible });
  }

  _handleVisibilityChange = (visible) => {
    this.setState({ visible });
  };

  render() {
    return (
      <section className="md-background--card theme-preview" ref={container => { this._container = container; }}>
        <Toolbar
          nav={<Button icon onClick={this._toggleDrawer}>menu</Button>}
          title="Theme Preview"
          colored
          fixed
        />
        <div className="md-grid md-toolbar-relative">
          <h2 className="md-display-1 md-cell md-cell--12">Look at this</h2>
          <LoremIpsum units="sentences" className="md-cell md-cell--12" component="p" />
          <Button primary raised style={{ marginLeft: 8, marginTop: '1em' }}>
            Button
          </Button>
        </div>
        <Drawer
          renderNode={this._container}
          header={<Toolbar title="Theme Preview" />}
          navClassName="md-toolbar-relative"
          navItems={NAV_ITEMS}
          visible={this.state.visible}
          onVisibilityChange={this._handleVisibilityChange}
          overlay
          type={Drawer.DrawerTypes.TEMPORARY}
        />
        <Button floating secondary fixed>email</Button>
      </section>
    );
  }
}
