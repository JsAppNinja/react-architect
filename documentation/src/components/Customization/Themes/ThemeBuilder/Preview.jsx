import React, { PureComponent } from 'react';
import Button from 'react-md/lib/Buttons/Button';
import Toolbar from 'react-md/lib/Toolbars';
import Drawer from 'react-md/lib/Drawers';

const NAV_ITEMS = ['Woop woop', 'That\'s the sound', 'Of the Police'];
const styles = {
  button: {
    marginLeft: 8,
    marginTop: '1em',
  },
};

export default class Preview extends PureComponent {
  constructor(props) {
    super(props);

    this.state = { visible: false };
  }

  toggleDrawer = () => {
    this.setState({ visible: !this.state.visible });
  };

  handleVisibilityChange = (visible) => {
    this.setState({ visible });
  };

  render() {
    return (
      <section className="md-background--card theme-preview" ref={(container) => { this._container = container; }}>
        <Toolbar
          nav={<Button icon onClick={this.toggleDrawer}>menu</Button>}
          title="Theme Preview"
          colored
          fixed
        />
        <div className="md-grid md-toolbar-relative">
          <h2 className="md-display-1 md-cell md-cell--12">Look at this</h2>
          <Button primary raised style={styles.button}>
            Button
          </Button>
        </div>
        <Drawer
          renderNode={this._container}
          header={<Toolbar title="Theme Preview" />}
          navClassName="md-toolbar-relative"
          navItems={NAV_ITEMS}
          visible={this.state.visible}
          onVisibilityChange={this.handleVisibilityChange}
          overlay
          type={Drawer.DrawerTypes.TEMPORARY}
        />
        <Button floating secondary fixed>email</Button>
      </section>
    );
  }
}
