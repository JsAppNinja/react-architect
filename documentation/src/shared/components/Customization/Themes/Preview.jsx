import React, { PureComponent, PropTypes } from 'react';
import Button from 'react-md/lib/Buttons/Button';
import Toolbar from 'react-md/lib/Toolbars';
import Drawer from 'react-md/lib/Drawers';

import './_styles.scss';
import LoremIpsum from 'components/LoremIpsum';
const NAV_ITEMS = [{ primaryText: 'Woop Woop' }, { primaryText: 'That\'s the Sound' }, { primaryText: 'Of the Police' }];

export default class Preview extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
  };

  constructor(props) {
    super(props);

    this.state = { visible: false };
    this._setContainer = this._setContainer.bind(this);
    this._toggleDrawer = this._toggleDrawer.bind(this);
    this._handleVisibilityToggle = this._handleVisibilityToggle.bind(this);
  }

  _setContainer(container) {
    this._container = container;
    if (this._container) {
      // trigger rerender
      this.setState({ found: true });
    }
  }

  _toggleDrawer() {
    this.setState({ visible: !this.state.visible });
  }

  _handleVisibilityToggle(visible) {
    this.setState({ visible });
  }

  render() {
    let drawer;
    if (this._container) {
      drawer = (
        <Drawer
          renderNode={this._container}
          header={<Toolbar title="Theme Preview" />}
          navClassName="md-toolbar-relative"
          navItems={NAV_ITEMS}
          visible={this.state.visible}
          onVisibilityToggle={this._handleVisibilityToggle}
          overlay
          type={Drawer.DrawerTypes.TEMPORARY}
        />
      );
    }

    return (
      <section className="md-background--card theme-preview" ref={this._setContainer}>
        <Toolbar
          nav={<Button icon onClick={this._toggleDrawer}>menu</Button>}
          title="Theme Preview"
          colored
          fixed
        />
        <div className="md-grid md-toolbar-relative">
          <h2 className="md-display-1 md-cell md-cell--12">Look at this</h2>
          <LoremIpsum units="sentences" className="md-cell md-cell--12" component="p" />
          <Button label="Button" primary raised style={{ marginLeft: 8, marginTop: '1em' }} />
        </div>
        {drawer}
        <Button floating secondary fixed>email</Button>
      </section>
    );
  }
}
