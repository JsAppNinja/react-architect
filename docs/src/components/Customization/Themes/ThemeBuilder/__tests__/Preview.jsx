/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Preview from '../Preview';

describe('Preivew', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Preview />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
