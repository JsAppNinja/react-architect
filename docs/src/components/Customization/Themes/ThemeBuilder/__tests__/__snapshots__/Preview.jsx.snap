// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview should render correctly 1`] = `
<section
  className="md-background--card theme-preview"
>
  <header
    className="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed"
    id="theme-builder-preview-toolbar"
    style={undefined}
  >
    <button
      className="md-inline-block md-btn md-btn--icon md-pointer--hover md-btn--toolbar md-toolbar--action-left"
      disabled={undefined}
      href={undefined}
      id="theme-builder-preview-drawer-toggle"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <i
        className="md-icon material-icons"
        style={undefined}
      >
        menu
      </i>
    </button>
    <h2
      className="md-title md-title--toolbar"
      id="theme-builder-preview-toolbar-title"
      style={undefined}
    >
      Theme Preview
    </h2>
  </header>
  <div
    className="md-grid md-toolbar-relative"
  >
    <h2
      className="md-display-1 md-cell md-cell--12"
    >
      Look at this
    </h2>
    <button
      className="md-inline-block md-btn md-btn--raised md-background--primary md-background--primary-hover md-pointer--hover md-btn--text theme-preview__btn"
      disabled={undefined}
      href={undefined}
      id="theme-builder-preview-btn"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Button
    </button>
  </div>
  <button
    className="md-inline-block md-btn md-btn--icon md-btn--floating md-background--secondary md-background--secondary-hover md-pointer--hover md-paper md-paper--2 md-btn--fixed md-btn--fixed-br"
    disabled={undefined}
    href={undefined}
    id="theme-builder-preview-floating-btn"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <i
      className="md-icon material-icons"
      style={undefined}
    >
      email
    </i>
  </button>
</section>
`;

exports[`Preview should render correctly after the drawer has become visible 1`] = `
<section
  class="md-background--card theme-preview"
>
  <span>
    <div
      class="md-overlay md-overlay--drawer md-pointer--hover md-overlay--active"
    />
  </span>
  <header
    class="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed"
    id="theme-builder-preview-toolbar"
  >
    <button
      class="md-inline-block md-btn md-btn--icon md-pointer--hover md-btn--toolbar md-toolbar--action-left"
      id="theme-builder-preview-drawer-toggle"
      type="button"
    >
      <i
        class="md-icon material-icons"
      >
        menu
      </i>
    </button>
    <h2
      class="md-title md-title--toolbar"
      id="theme-builder-preview-toolbar-title"
    >
      Theme Preview
    </h2>
  </header>
  <div
    class="md-grid md-toolbar-relative"
  >
    <h2
      class="md-display-1 md-cell md-cell--12"
    >
      Look at this
    </h2>
    <button
      class="md-inline-block md-btn md-btn--raised md-background--primary md-background--primary-hover md-pointer--hover md-btn--text theme-preview__btn"
      id="theme-builder-preview-btn"
      type="button"
    >
      Button
    </button>
  </div>
  <nav
    class="md-paper md-paper--5 md-drawer md-drawer--left md-drawer--fixed md-drawer--active md-transition--decceleration md-background--card"
    id="theme-builder-preview-drawer"
  >
    <header
      class="md-paper md-paper--0 md-toolbar"
      id="theme-builder-preview-drawer-toolbar"
    >
      <h2
        class="md-title md-title--toolbar"
        id="theme-builder-preview-drawer-toolbar-title"
      >
        Theme Preview
      </h2>
    </header>
    <ul
      class="md-list md-list--drawer md-toolbar-relative"
      id="theme-builder-preview-drawer-nav-items"
    >
      <li
        class="md-list-item"
      >
        <div
          aria-pressed="false"
          class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-list-tile md-text"
          role="button"
          tabindex="0"
        >
          <div
            class="md-tile-content"
          >
            <div
              class="md-tile-text--primary md-text"
            >
              Woop woop
            </div>
          </div>
        </div>
      </li>
      <li
        class="md-list-item"
      >
        <div
          aria-pressed="false"
          class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-list-tile md-text"
          role="button"
          tabindex="0"
        >
          <div
            class="md-tile-content"
          >
            <div
              class="md-tile-text--primary md-text"
            >
              That's the sound
            </div>
          </div>
        </div>
      </li>
      <li
        class="md-list-item"
      >
        <div
          aria-pressed="false"
          class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-list-tile md-text"
          role="button"
          tabindex="0"
        >
          <div
            class="md-tile-content"
          >
            <div
              class="md-tile-text--primary md-text"
            >
              Of the Police
            </div>
          </div>
        </div>
      </li>
    </ul>
  </nav>
  <button
    class="md-inline-block md-btn md-btn--icon md-btn--floating md-background--secondary md-background--secondary-hover md-pointer--hover md-paper md-paper--2 md-btn--fixed md-btn--fixed-br"
    id="theme-builder-preview-floating-btn"
    type="button"
  >
    <i
      class="md-icon material-icons"
    >
      email
    </i>
  </button>
</section>
`;
